docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,438 @@
+    // associated with the given protocol.ID.
+    BandwidthForProtocol(proto protocol.ID) Stats
+   
+    // ResourceState returns the state of the resource manager.
+    ResourceState() rcmgr.ResourceManagerStat
+   
+    // PubSubPeers returns the peer IDs of the peers joined on
+    // the given topic.
+    PubSubPeers(topic string) ([]peer.ID, error)
+  }
+```
+
+### NodeModule
+
+```go
+
+  type NodeModule interface {
+    // Type returns the node type.
+    Type() node.Type
+    // Version returns information about the current binary build.
+    Version() string
+   
+    // LogLevelSet sets the given component log level to the given level.
+    LogLevelSet(ctx context.Context, name, level string) error
+  }
+  
+```
+
+#### DAS
+
+```go
+  type DASModule interface {
+    // Stats returns current stats of the DASer.
+    Stats() (das.SamplingStats, error)
+  }
+```
+
+#### State
+
+```go
+  type StateModule interface {
+    // AccountAddress retrieves the address of the node's account/signer
+    AccountAddress(ctx context.Context) (state.Address, error)
+    // Balance retrieves the Celestia coin balance for the node's account/signer
+    // and verifies it against the corresponding block's AppHash.
+    Balance(ctx context.Context) (*state.Balance, error)
+    // BalanceForAddress retrieves the Celestia coin balance for the given 
+    // address and verifies the returned balance against the corresponding
+    // block's AppHash.
+    BalanceForAddress(ctx context.Context, addr state.Address) (*state.Balance, error)
+    // SubmitTx submits the given transaction/message to the Celestia network
+    // and blocks until the tx is included in a block.
+    SubmitTx(ctx context.Context, tx state.Tx) (*state.TxResponse, error)
+    // SubmitPayForData builds, signs and submits a PayForData transaction.
+    SubmitPayForData(
+      ctx context.Context, 
+      nID namespace.ID, 
+      data []byte, 
+      gasLimit uint64,
+    ) (*state.TxResponse, error)
+    // Transfer sends the given amount of coins from default wallet of the node 
+    // to the given account address.
+    Transfer(
+      ctx context.Context, 
+      to types.Address, 
+      amount types.Int, 
+      gasLimit uint64,
+    ) (*state.TxResponse, error)
+
+    // StateModule also provides StakingModule
+    StakingModule
+  }
+```
+
+Ideally all the state modules below should be implemented on top of only
+StateModule, but there is no way we can have an abstract state requesting method,
+yet.
+
+##### Staking
+
+```go
+  type StakingModule interface {
+    // Delegate sends a user's liquid tokens to a validator for delegation.
+    Delegate(
+        ctx context.Context, 
+        delAddr state.ValAddress, 
+        amount state.Int, 
+        gasLim uint64,
+    ) (*state.TxResponse, error)
+    // BeginRedelegate sends a user's delegated tokens to a new validator for redelegation.
+    BeginRedelegate(
```

Comment:
nit: why no just `Redelegate`?

Reply:
We keep the naming scheme of celestia-core